data/core/commandview.lua

```diff
@@ -348,7 +348,7 @@ function CommandView:draw_line_gutter(idx, x, y)
   local color = common.lerp(style.text, style.accent, self.gutter_text_brightness / 100)
   core.push_clip_rect(pos.x, pos.y, self:get_gutter_width(), self.size.y)
   x = x + style.padding.x
-  renderer.draw_text(self:get_font(), self.label, x, y + yoffset, color)
+  renderer.draw_text(core.window, self:get_font(), self.label, x, y + yoffset, color)
```

Comment:
a discussion around this API change already happened on discord, so I'm marking this to get more feedback.

The current implementation requires that the first argument is the relevant window, but this change would be backwards incompatible.

Alternatives include:
- making window argument optional (checking arg count, putting it at the end, etc.)
    + this would make adding arguments more painful in the future
- turning the window into a global
    + Personally I am not the biggest fan of it because its likely going to introduce race conditions if done wrong.
    + The only way I could see this being done is by having a set and unset (push and pop?) function and erroring out if you are trying to set when its already unset
- leaving the existing draw functions as is and adding `renwindow:draw_*` functions which act directly on the relevant window
     + the `renderer.draw_*` methods could then be redirected to the main window methods
- accept that window is an argument of the draw function
    + this is way more explicit and ensures that intended behavior